sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/NestedColumnAliasing.scala

```diff
@@ -107,16 +107,17 @@ object NestedColumnAliasing {
    * 1. ExtractValue -> Alias: A new alias is created for each nested field.
    * 2. ExprId -> Seq[Alias]: A reference attribute has multiple aliases pointing it.
    */
-  def getAliasSubMap(exprList: Seq[Expression])
+  def getAliasSubMap(exprList: Seq[Expression], exclusiveAttrs: Seq[Attribute] = Seq.empty)
```

Comment:
Ur, I see. This is a nit comment though, I feel sharing `getAliasSubMap` in both cases (`project w/generate case` and `simple project case`) makes the function arguments a bit complicated. So, its better to leave some comments about the argument `exclusiveAttrs` in the line 105.

Reply:
Thanks. Added few comments.